react_juce/core/Utils.cpp

```diff
@@ -22,5 +22,154 @@ namespace reactjuce
+                x2 = edgePoint2[0]; y2 = edgePoint2[1];
+                double gradientLineDistance = abs(b.getWidth() * sin(radians) + abs(b.getHeight() * cos(radians)));
+                double edgePoint1CenterDist = sqrt(pow((x1 - centerPoint[0]), 2) + pow((y1 - centerPoint[1]), 2));
+                double edgePoint2CenterDist = sqrt(pow((x2 - centerPoint[0]), 2) + pow((y2 - centerPoint[1]), 2));
+                double halfGradientDistance = gradientLineDistance/2;
+                double distanceRatio1 = halfGradientDistance/edgePoint1CenterDist;
+                double distanceRatio2 = halfGradientDistance/edgePoint2CenterDist;
+                x1 = ((1 - distanceRatio1) * centerPoint[0]) + (distanceRatio1 * x1);
+                y1 = ((1 - distanceRatio1) * centerPoint[1]) + (distanceRatio1 * y1);
+                x2 = ((1 - distanceRatio2) * centerPoint[0]) + (distanceRatio2 * x2);
+                y2 = ((1 - distanceRatio2) * centerPoint[1]) + (distanceRatio2 * y2);
+                //Gradient line Points
+                int glX1 = x1; int glX2 = x2; int glY1 = y1; int glY2 = y2;
+                //Offset the gradient line length with min max extension percent
+                x1 = ((1 - (negativeExtensionPercent * -1)) * glX1) + ((negativeExtensionPercent * -1) * glX2);
+                y1 = ((1 - (negativeExtensionPercent * -1)) * glY1) + ((negativeExtensionPercent * -1) * glY2);
+                x2 = ((1 - (positiveExtensionPercent * -1)) * glX2) + ((positiveExtensionPercent * -1) * glX1);
+                y2 = ((1 - (positiveExtensionPercent * -1)) * glY2) + ((positiveExtensionPercent * -1) * glY1);
+                juce::ColourGradient gradient = juce::ColourGradient (minColor, x1, y1, maxColor, x2, y2,false);
+                for (auto& colors : colorStopsObj->getProperties())
+                {
+                    juce::String colorID = colors.name.toString();
+                    auto* color = colorStopsObj->getProperty(colorID).getDynamicObject();
+                    //Calculate any offset from the original additional color percentage along gradient line
+                    if(color->getProperty("type") == "additional"){
+                        juce::String hexString = color->getProperty("hex");
+                        juce::Colour additionalColor = juce::Colour::fromString(hexString);
+                        double originalPercent = color->getProperty("position");
+                        double gradientLineDistanceExtension = gradientLineDistance * (1 + (negativeExtensionPercent + positiveExtensionPercent));
+                        double offsetPercent = ((gradientLineDistance * originalPercent) + (gradientLineDistance * negativeExtensionPercent)) / gradientLineDistanceExtension;
+                        gradient.addColour(offsetPercent, additionalColor);
+                    }
+                }
+                colorVar = gradient;
```

Comment:
Maybe @nick-thompson or @JoshMarler can confirm, I think in the to-be written style guide returning inside the if would be preferred to adding an else.

i.e.
```
// no declaration of colorVar
if(colorVariant.isObject()) {
  // ...
  juce::ColourGradient gradient = juce::ColourGradient (minColor, x1, y1, maxColor, x2, y2,false);
  // ...
  return gradient
}

return juce::Colour::fromString(colorVariant.toString());
```

Almost not worth mentioning, but the style guide needs a little love.  Also apologies if I've got it completely the wrong way around.

Reply:
O yup sweet as think I see a few other instances of this style. I'll change that

Reply:
O yup sweet as think I see a few other instances of this style. I'll change that